web/src/toga_web/app.py

```diff
@@ -40,25 +43,17 @@ def create(self):
 
     def _create_submenu(self, group, items):
         submenu = create_element(
-            "li",
```

Comment:
With both Chrome and Safari, the first time I choose the About menu, something flickers on the left side of the screen, but the dialog doesn't appear. Choosing the menu a second time works OK.

Reply:
I saw this... but thought I'd fixed it. Turns out, Safari was just so good at caching that it wasn't happening after my initial successful load.

The underlying issue is that Shoelace autoloads elements when they first occur. Since the page doesn't contain an `<sl-dialog>`, the element is added to the DOM, and appears unstyled. Shoelace responds by autoloading the element (which hides the dialog content); but at this point, it's too late, and the `show()` call has already been invoked  and the dialog isn't displayed. On second run, the `<sl-dialog>` DOM element already exists, so the dialog appears.

However, if we defer the `show()` call until the Promise from the load of the `<sl-dialog>` element is fulfilled, the problem goes away. On first display, there's a tiny delay as the element is constructed; on second display, the Promise is immediately fulfilled, and the dialog is displayed immediately.